Calculate results and model snapshot retention using latest bucket timestamps (#51061)
elastic/elasticsearch: docs/reference/ml/ml-shared.asciidoc

end::model-snapshot-id[]

tag::model-snapshot-retention-days[]

> Advanced configuration option. Denotes the period for which model snapshots

Author: @szabosteve can you look over these docs changes please. Do they make sense?

Reviewer: @davidkyle Sorry, I haven't noticed this one earlier.

> Advanced configuration option. Denotes the period for which model snapshots

Reviewer 2: If this property is still a number that represents days, I think it's worth keeping that in the definition.  For example:
```suggestion
Advanced configuration option. The period of time (in days) that model snapshots are retained.
```

Author: Yes that's better thanks

> will be retained. Age is calculated as the time from the newest model snapshot's

Reviewer 2: Not mandatory, but I try to avoid possessives:
```suggestion
Age is calculated relative to the timestamp of the newest model snapshot.
```

timestamp, older snapshots are automatically deleted. The default value is `1`,

> which means snapshots are retained for one day (twenty-four hours).

Reviewer 2: Maybe move the deletion info here?:
```suggestion
which means snapshots that are one day (twenty-four hours) older than the newest snapshot are deleted.
```

end::results-index-name[]

tag::results-retention-days[]

> Advanced configuration option. When set this option will prune results older

Reviewer 2: I'm not sure I understand why we're making the descriptions diverge like this (focus on pruning vs retaining). I'd recommend keeping them in synch unless I'm missing something important here. For example:

```suggestion
Advanced configuration option. The period of time (in days) that results are retained.
```

> than this age in days. Age is calculated as the difference between the latest

Reviewer 2: Note sure if newest or latest is correct, bit I think it should be used similarly in both definitions:
```suggestion
Age is calculated relative to the timestamp of the latest bucket result.
```

> bucket result's timestamp and the result's timestamp. Once per day at 00:30

Reviewer 2: Moving the "when set this option..." here:
```suggestion
If this property has a non-null value, once per day at 00:30
```

> (server time), results older than this are deleted from {es}. The default

Reviewer 2: Trying to match suggestion in the other definition :) :
```suggestion
(server time), results that are the specified number of days older than the newest result are deleted from {es}. The default
```
